lighthouse-core/test/config/config-test.js

```diff
@@ -147,6 +147,74 @@ describe('Config', () => {
     })).toThrow('VRMLElements gatherer, required by audit missing-artifact-audit, was not found in config');
   });
 
+  // eslint-disable-next-line max-len
+  it('does not throw when an audit requests an optional artifact with no gatherer supplying it', async () => {
+    class DoesntNeedYourCrap extends Audit {
+      static get meta() {
+        return {
+          id: 'optional-artifact-audit',
+          title: 'none',
+          description: 'none',
+          requiredArtifacts: [
+            'URL', // base artifact
+            'ViewportDimensions', // from gatherer
+          ],
+          optionalArtifacts: [
+            'SourceMaps', // Not in the config.
+          ],
+        };
+      }
+
+      static audit() {}
+    }
+
+    // Shouldn't throw.
+    const config = new Config({
+      passes: [{
+        passName: 'defaultPass',
+        gatherers: [
+          'viewport-dimensions',
+        ],
+      }],
+      audits: [DoesntNeedYourCrap],
+    });
+    expect(config.passes[0].gatherers.map(g => g.path)).toEqual(['viewport-dimensions']);
+  });
+
+  it('does warn when an audit requests and receives an optional artifact', async () => {
```

Comment:
```suggestion
  it('should keep optionalArtifacts in gatherers after filter', async () => { 
```

or something? the warn is just copy/paste cruft right

Reply:
no just brain fart :)